pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,628 @@
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return vmexport, nil
+}
+
+// createVirtualMachineExport serves as a wrapper to create the virtualMachineExport object and, if needed, do error handling
+func createVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	vmexport, err := getVirtualMachineExport(client, namespace)
+	if err != nil {
+		return err
+	}
+	if vmexport != nil {
+		return fmt.Errorf("VirtualMachineExport '%s/%s' already exists", namespace, vmexportName)
+	}
+
+	vmexport = &exportv1.VirtualMachineExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmexportName,
+			Namespace: namespace,
+		},
+		Spec: exportv1.VirtualMachineExportSpec{
+			TokenSecretRef: getExportSecretName(),
+			Source:         exportSource,
+		},
+	}
+
+	vmexport, err = client.VirtualMachineExport(namespace).Create(context.TODO(), vmexport, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Generate/get secret to be used with the vmexport
+	_, err = getOrCreateTokenSecret(client, vmexport)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' created succesfully\n", namespace, vmexportName)
+	return nil
+}
+
+// deleteVirtualMachineExport serves as a wrapper to delete the virtualMachineExport object
+func deleteVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if err := client.VirtualMachineExport(namespace).Delete(context.TODO(), vmexportName, metav1.DeleteOptions{}); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		return fmt.Errorf("VirtualMachineExport '%s/%s' does not exist", namespace, vmexportName)
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' deleted succesfully\n", namespace, vmexportName)
+	return nil
+}
+
+// downloadVirtualMachineExport handles the process of downloading the requested volume from a VirtualMachineExport object
+func downloadVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if shouldCreate {
+		if err := createVirtualMachineExport(client, namespace); err != nil {
+			return err
+		}
+	}
+
+	if !keepvme {
+		defer deleteVirtualMachineExport(client, namespace)
+	}
+
+	// Wait for the vmexport object to be ready
+	if err := waitForVirtualMachineExport(client, namespace, processingWaitInterval, processingWaitTotal); err != nil {
+		return err
+	}
+
+	vmexport, err := getVirtualMachineExport(client, namespace)
+	if err != nil {
+		return err
+	}
+	if vmexport == nil {
+		return fmt.Errorf("Unable to get '%s/%s' VirtualMachineExport", namespace, vmexportName)
+	}
+
+	// Download the exported volume
+	if err := downloadVolume(client, vmexport); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadVolume handles the process of downloading the requested volume from a VirtualMachineExport
+func downloadVolume(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) error {
+	output, err := os.Create(outputFile)
```

Comment:
I think if the download fails in any reason we should delete the created outputfile, it will be either corrupt or empty and it is confusing that we have an output if the download failed (confused me trying it localy)

Reply:
Yeah, I think a good middle point it's to move the `os.Create` just before the copying starts, so we don't leave an empty file in case the request or the vmexport fails.